include/Link.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+
+class Link {
+public:
+    Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t distance_ns);
+
+    void UpdateLastProcessTime(std::uint64_t new_process_time_ns);
```

Comment:
A more grammatically correct name for a variable here is `new_processing_time_ns`.

```suggestion
    void UpdateLastProcessTime(std::uint64_t new_processing_time_ns);
```